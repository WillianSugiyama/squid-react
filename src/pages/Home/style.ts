import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  height: 100vh;
  width: 100%;
  background-color: black;
  color: white;
  align-items: center;
  justify-content: center;
  flex-direction: column;

  img {
    height: 50%;
  }

  h1 {
    margin-top: 5%;
  }
`;
