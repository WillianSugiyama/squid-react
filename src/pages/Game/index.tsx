import React, { useState } from 'react'
import { useNavigate } from "react-router-dom";

import { Container, GameContainer, GameEndContainer } from "./style"
import { List } from '../../components/List'
import { Button } from '../../components/Button'

import NumberFormat from 'react-number-format'
import { generateFake } from '../../utils/generateFakePersons';
import { calcPercentage } from '../../utils/calcPercentage';
import { Person } from '../../interfaces/person';

const listFake = generateFake();

export const GamePage = () => {
  const history = useNavigate()
  const [round, setRound] = useState(1)
  const [budget, setBudget] = useState(0.00)
  const [endGameCount, setEndGameCount] = useState(0)
  const [persons, setPersons] = useState<Person[]>(listFake)
  const [deadPersons, setDeadPersons] = useState<Person[]>([])
  const [gameVisibility, setGameVisibility] = useState(true)

  const startGame = () => {
    const audio = new Audio()
    audio.src = '/public_red-light-sound.mp3'
    audio.play()

    setTimeout(() => {
      audio.pause()
      audio.currentTime = 0;

      const calculedPerson: Person[] = persons.map((person: Person) => {
        return {
          name: person.name,
          isActive: calcPercentage(42),
          isQuitter: calcPercentage(30)
        }
      })

      const exitedPersons = [...deadPersons, ...calculedPerson.filter((person: Person) => !person.isActive)]
      const personsRemaining = calculedPerson.filter((person: Person) => person.isActive)
      const valueRemaining = exitedPersons.length * 420000
      const endGameVotes = personsRemaining.filter((person) => person.isQuitter).length

      personsRemaining.length <= 1 ? setGameVisibility(false): setGameVisibility(true)
      endGameVotes >= (personsRemaining.length / 2) ? setGameVisibility(false): setGameVisibility(true)

      setPersons(personsRemaining)
      setDeadPersons(exitedPersons)
      setRound(round + 1)
      setBudget(valueRemaining)
      setEndGameCount(endGameVotes)
    }, 8000)
  }

  const endGame = (navigate: any) => {
    navigate('/')
  }


  return (
    <Container>
      <List headerText='Jogadores Remanescentes: ' persons={persons} />
      <GameContainer display={gameVisibility}>
        <h1> Round: {round} </h1>
        <h2> Fundo do premio: <NumberFormat value={budget} prefix={'$'} thousandSeparator={true} displayType={'text'} /></h2>
        <img src='/red-man.gif'/>
        <h2> Votos para o fim do jogo: {endGameCount}</h2>
        <Button trigger={() => {startGame()}}> Iniciar Rodada </Button>
      </GameContainer>
      <GameContainer display={!gameVisibility}>
        <GameEndContainer>
          <p> Total de Jogadores Remanescentes: {persons.length} </p>
          <p> Total de Jogadores Eliminados: {deadPersons.length} </p>
          <p> Votos para o fim do jogo: {endGameCount} </p>
          <p> Premio por jogador Remanescente: <NumberFormat value={budget / persons.length} prefix={'$'} thousandSeparator={true} displayType={'text'}/> </p>
        </GameEndContainer>
        <img src='/red-man.gif'/>
        <Button trigger={() => {endGame(history)}}> Finalizar </Button>
      </GameContainer>
      <List headerText='Jogadores Eliminados:' persons={deadPersons} />
    </Container>
  )
}