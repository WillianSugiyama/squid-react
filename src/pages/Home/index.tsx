import React from 'react';

import { Container } from './style';
import { Button } from '../../components/Button';
import { useNavigate } from 'react-router-dom';

const handleButton = (navigate: any) => {
  const audio = new Audio();
  audio.src = '/public_main-theme.mp3';
  audio.play();

  setTimeout(() => {
    audio.pause();
    audio.currentTime = 0;
    navigate('/game');
  }, 3000);
};

export const HomePage = () => {
  const navigate = useNavigate();

  return (
    <Container>
      <img src='/squid-game-image.jpg' alt='JOGO DA LULA' />
      <h1> Jogo da Lula </h1>
      <Button trigger={() => handleButton(navigate)}>Come√ßar</Button>
    </Container>
  );
};
