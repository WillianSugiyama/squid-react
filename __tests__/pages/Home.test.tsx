import * as React from "react";
import { shallow } from "enzyme";

import { HomePage } from '../../src/pages/Home'

const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
   ...jest.requireActual('react-router-dom') as any,
  useNavigate: () => mockedUsedNavigate,
}));

it('Render HomePage', () => {
  const wrapper = shallow(<HomePage />)

  expect(wrapper).toMatchSnapshot()
})