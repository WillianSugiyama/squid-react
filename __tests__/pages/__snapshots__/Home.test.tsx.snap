// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render HomePage 1`] = `ShallowWrapper {}`;
